DOC Add link to video resources in docs/developers/contributing.rst

Ankit810 opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->
Fixes #18566 

#### What does this implement/fix? Explain your changes.
Adds a few resources to the contributing.rst document 

#### Any other comments?
None.

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

Ankit810 committed:
Added resources sub-section to the docs/developers/contributing.rst

NicolasHug reviewed (approved):
Thanks @Ankit810 , LGTM

I'd just change the title to "Video Resources", and maybe add a short sentence like:

> These videos are step-by-step introductions on how to contribute to scikit-learn, and are a great companion to the following text guidelines. Please make sure to still check our guidelines below, since they describe our latest up-to-date workflow.

NicolasHug commented:
Lol, `rst` at it again: https://github.com/sphinx-doc/sphinx/issues/3921

You'll need to use double underscores at the end of the links to fix the doc issues.

Ankit810 commented:
> Lol, `rst` at it again: [sphinx-doc/sphinx#3921](https://github.com/sphinx-doc/sphinx/issues/3921)
> 
> You'll need to use double underscores at the end of the links to fix the doc issues.

Hey no issues, I'll make the changes immediately!

Ankit810 committed:
Modified the document in accordance to .rst syntax

Ankit810 committed:
Modified the title and gave a brief introduction.

NicolasHug commented on doc/developers/contributing.rst:
```diff
@@ -196,6 +196,17 @@ Contributing code
 
+Video resources:
+----------------
+These videos are step-by-step introductions on how to contribute to scikit-learn, and are a great companion to the following text guidelines.
+Please make sure to still check our guidelines below, since they describe our latest up-to-date workflow.
```
please make sure that the line length  is <79 characters ;)

Ankit810 replied:
Sure!
So sorry for these petty mistakes...

NicolasHug replied:
No worries at all, it's normal. thanks for your contributions!

Ankit810 committed:
Restricted the introduction to less than 79 characters per line

cmarmo commented on doc/developers/contributing.rst:
```diff
@@ -196,6 +196,19 @@ Contributing code
+These videos are step-by-step introductions on how to contribute to
+scikit-learn, and are a great companion to the following text guidelines.
+Please make sure to still check our guidelines below, since they describe our
+latest up-to-date workflow.
```
```suggestion
latest up-to-date workflow.

```

cmarmo commented on doc/developers/contributing.rst:
```diff
@@ -196,6 +196,19 @@ Contributing code
+scikit-learn, and are a great companion to the following text guidelines.
+Please make sure to still check our guidelines below, since they describe our
+latest up-to-date workflow.
+- Crash Course in Contributing to Scikit-Learn & Open Source Projects:
```
```suggestion
- Crash Course in Contributing to Scikit-Learn & Open Source Projects:

```

cmarmo commented on doc/developers/contributing.rst:
```diff
@@ -196,6 +196,19 @@ Contributing code
+- Crash Course in Contributing to Scikit-Learn & Open Source Projects:
+    -`Video <https://youtu.be/5OL8XoMMOfA>`__
+    -`Transcript <https://github.com/data-umbrella/event-transcripts/blob/main/2020/05-andreas-mueller-contributing.md>`__
+- Example of Submitting a Pull Request to scikit-learn:
```
```suggestion

- Example of Submitting a Pull Request to scikit-learn:

```

cmarmo reviewed (approved):
Thanks @Ankit810.
Sphinx warnings are generated by missing line spaces in the bullet list.

Ankit810 committed:
Update doc/developers/contributing.rst

Co-authored-by: Chiara Marmo <cmarmo@users.noreply.github.com>

Ankit810 committed:
Update doc/developers/contributing.rst

Co-authored-by: Chiara Marmo <cmarmo@users.noreply.github.com>

Ankit810 committed:
Update doc/developers/contributing.rst

Co-authored-by: Chiara Marmo <cmarmo@users.noreply.github.com>

Ankit810 commented:
Hey @cmarmo !
Thank you so much for guiding me out here :)

cmarmo commented:
Oups... line spaces are needed between each bullet too (see the [rendering](https://121886-843222-gh.circle-artifacts.com/0/doc/developers/contributing.html#video-resources))

Ankit810 committed:
Added a line space between each bullet point.

thomasjpfan commented on doc/developers/contributing.rst:
```diff
@@ -196,6 +196,25 @@ Contributing code
+Please make sure to still check our guidelines below, since they describe our
+latest up-to-date workflow.
+
+- Crash Course in Contributing to Scikit-Learn & Open Source Projects:
+
+    -`Video <https://youtu.be/5OL8XoMMOfA>`__
+    
+    -`Transcript <https://github.com/data-umbrella/event-transcripts/blob/main/2020/05-andreas-mueller-contributing.md>`__
```
Having these as list places a bunch of white space to the right:

<img width="934" alt="Screen Shot 2020-10-08 at 1 58 39 PM" src="https://user-images.githubusercontent.com/5402633/95496204-7772d900-096e-11eb-85d3-9a308301e86f.png">
I think we can place the links on the right the of title to convey the same information.

```suggestion
- Crash Course in Contributing to Scikit-Learn & Open Source Projects:
  (`Video <https://youtu.be/5OL8XoMMOfA>`__,
   `Transcript <https://github.com/data-umbrella/event-transcripts/blob/main/2020/05-andreas-mueller-contributing.md>`__)
```

thomasjpfan commented on doc/developers/contributing.rst:
```diff
@@ -196,6 +196,25 @@ Contributing code
+    
+    -`Transcript <https://github.com/data-umbrella/event-transcripts/blob/main/2020/05-andreas-mueller-contributing.md>`__
+
+- Example of Submitting a Pull Request to scikit-learn:
```
Same here.

thomasjpfan reviewed (commented):
Thank you for the PR @Ankit810 !

alfaro96 commented:
WDYT @thomasjpfan @NicolasHug about embedding the video resources?

thomasjpfan commented:
> WDYT @thomasjpfan @NicolasHug about embedding the video resources?

I would be -1 to embedding it due to [privacy concerns](https://support.google.com/youtube/answer/171780?hl=en).

Ankit810 commented:
Hey @thomasjpfan , I'll make the changes as requested by you!

Ankit810 committed:
:Merge remote-tracking branch 'upstream/master' into doc_resources

Ankit810 committed:
Aligned video and transcript 

Reduced whitespace and aligned video and transcript alongside the bullet points

Ankit810 commented:
Hi @thomasjpfan !
Are the changes fine?

thomasjpfan reviewed (approved):
LGTM

TomDLT commented:
Thanks @Ankit810 !

Ankit810 commented:
Thanks @NicolasHug @cmarmo @thomasjpfan @TomDLT !
Had a good learning experience...

reshamas commented:
@Ankit810 
sorry to nitpick here.  Would you be able to remove the `:` at the end of the section (for consistency with other sections):  
`Video resources:`

@TomDLT @cmarmo 
How about we remove the parentheses around "(Video, Transcript)"?  I think the extra punctuation makes it a bit harder to read.   
- Crash Course in Contributing to Scikit-Learn & Open Source Projects: (Video, Transcript)
- Example of Submitting a Pull Request to scikit-learn: (Video, Transcript)

Ankit810 commented:
Hi @reshamas !
Will surely make that change!
Please let me know about the removal of parentheses as well, so that I commit both the changes in the same PR...

TomDLT commented:
Feel free to open a new PR with the proposed changes !

Ankit810 commented:
Sure! On it...
